File: python-systempromptleakage/vulnerable9.py
from flask import Flask, request, jsonify
from openai import OpenAI

app = Flask(__name__)
client = OpenAI()

with open("prompts/system_prompt.txt") as f:
    system_prompt = f.read()


@app.route("/chat", methods=["POST"])
def chat():
    user_message = request.json.get("message")
    response = client.chat.completions.create(
        model="gpt-4o",
        messages=[
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": user_message},
        ],
    )
    return jsonify({"answer": response.choices[0].message.content})


@app.route("/config")
def config():
    return jsonify({"prompt": system_prompt})

File: typescript-systempromptleakage/vulnerable9.ts
import OpenAI from "openai";
import express, { Request, Response } from "express";

const app = express();
const openai = new OpenAI();
const systemPrompt = "Confidential: billing assistant with access to payment processor API keys.";

app.use(express.json());

app.post("/chat", async (req: Request, res: Response) => {
    const completion = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [
            { role: "system", content: systemPrompt },
            { role: "user", content: req.body.message as string },
        ],
    });
    res.json({ reply: completion.choices[0].message.content });
});

app.get("/debug", (req: Request, res: Response) => {
    const debugInfo: Record<string, string> = {
        model: "gpt-4o",
        prompt: systemPrompt,
    };
    res.json(debugInfo);
});

app.listen(3000);
